### PR TITLE
EA-757 sample  data connection data from mock sdk-ProviderSearchRequest-searchConnections-searchProviders

### PR DESCRIPTION
**Pull request by Arunkumarbwell**
Updated SDK version 0.3.0,integrated Data connection flow :- ProviderSearchRequest(),searchConnections(),searchProviders()

### CODE REVIEW

File: bwell-kotlin-android/app/src/main/java/com/bwell/sampleapp/activities/ui/data_connections/DataConnectionsFragment.kt [hunk 1]
```diff
@@ -296,12 +182,30 @@ class DataConnectionsFragment : Fragment(), View.OnClickListener, PopupFragment.
                         e.printStackTrace()
                     }
                 }
+                viewLifecycleOwner.lifecycleScope.launch {
```
Comment (istashak) [inline bwell-kotlin-android/app/src/main/java/com/bwell/sampleapp/activities/ui/data_connections/DataConnectionsFragment.kt:185]: I noticed that clicking "Disconnect" on the Lab connection doesn't do anything. I'd expect it to either disappear or change from Connected to Disconnected.

I'm not sure if this was part of the acceptance criteria, if not ignore this comment.

File: bwell-kotlin-android/app/src/main/java/com/bwell/sampleapp/utils/SelectedOrganizationHolder.kt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.bwell.sampleapp.utils
+
+import com.bwell.search.ProviderSearchQuery
+
+object SelectedOrganizationHolder {
+    var selectedOrganization: ProviderSearchQuery.Organization? = null
+}
```
Comment (istashak) [inline bwell-kotlin-android/app/src/main/java/com/bwell/sampleapp/utils/SelectedOrganizationHolder.kt:7]: This is an error prone way of holding the selectedOrganization.

I'd do this to the Fragment constructors that require it instead:

class OrganizationInfoFragment constructor(organization: ProviderSearchQuery.Organization) : Fragment() ...

File: bwell-kotlin-android/app/build.gradle.kts [hunk 3]
```diff
@@ -55,6 +55,7 @@ dependencies {
 
     // BWell SDK Usage
     implementation("com.bwell:bwell-sdk-kotlin:0.3.1")
+    implementation("com.apollographql.apollo3:apollo-api:3.8.2")
```
[the remark this answers is not in the document]
Comment (Arunkumarbwell) [reply bwell-kotlin-android/app/build.gradle.kts:58]: Yes tried to remove this but getting issue

[the remark this answers is not in the document]
Comment (Arunkumarbwell) [reply bwell-kotlin-android/app/build.gradle.kts]: Done

File: bwell-kotlin-android/app/src/main/java/com/bwell/sampleapp/utils/SelectedOrganizationHolder.kt (hunk 2 above)
Comment (Arunkumarbwell) [reply bwell-kotlin-android/app/src/main/java/com/bwell/sampleapp/utils/SelectedOrganizationHolder.kt]: Done

**Review by istashak — APPROVED** [review]

**Review by wfieldx — CHANGES_REQUESTED** [review]
make sure you're not checking in things that aren't to be checked in. e.g. local properties:

```
-## This file must *NOT* be checked into Version Control Systems,
-# as it contains information specific to your local configuration.
-#
-# Location of the SDK. This is only used by Gradle.
-# For customization when using a Version Control System, please read the
-# header note.
-#Thu Nov 02 03:15:32 GMT 2023
-sdk.dir=/Users/arunkumar/Library/Android/sdk
```

**Review by wfieldx — CHANGES_REQUESTED** [review]
1. User's name should be displayed here: Welcome!!BWell
2. navigation around data connections is incorrect (i.e. no back out of search, you have to select something other than "data connections"  from the side bar and go through it again)

**Comment by Arunkumarbwell** [comment]
> 1. User's name should be displayed here: Welcome!!BWell
> 2. navigation around data connections is incorrect (i.e. no back out of search, you have to select something other than "data connections"  from the side bar and go through it again)

Done

**Review by wfieldx — APPROVED** [review]

**Pull request merged**